### ISSUE TITLE
IndexError when inserting documents

### ISSUE THREAD

**Issue by johnymontana** [neo4j-contrib/neo4j_doc_manager#18]
When using `mongoimport` to insert [this sample data](https://raw.githubusercontent.com/mongodb/docs-assets/primer-dataset/dataset.json), an `IndexError` occurs and only 8434 documents (out of 25359 documents that were successfully inserted into MongoDB) are imported to Neo4j.

Stacktrace:

~~~
bash-3.2$ mongo-connector -m localhost:27017 -t http://localhost:7474/db/data -d neo4j_doc_manager
Logging to mongo-connector.log.
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/Library/Python/2.7/site-packages/mongo_connector/util.py", line 85, in wrapped
    func(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/mongo_connector/oplog_manager.py", line 256, in run
    docman.upsert(doc, ns, timestamp)
  File "/Users/lyonwj/neotechnology/devtmp/neo4j_doc_manager/mongo_connector/doc_managers/neo4j_doc_manager.py", line 59, in upsert
    builder = NodesAndRelationshipsBuilder(doc, doc_type, doc_id)
  File "/Users/lyonwj/neotechnology/devtmp/neo4j_doc_manager/mongo_connector/doc_managers/nodes_and_relationships_builder.py", line 12, in __init__
    self.build_nodes_query(doc_type, doc, doc_id)
  File "/Users/lyonwj/neotechnology/devtmp/neo4j_doc_manager/mongo_connector/doc_managers/nodes_and_relationships_builder.py", line 22, in build_nodes_query
    self.build_nodes_query(key, document[key], id)
  File "/Users/lyonwj/neotechnology/devtmp/neo4j_doc_manager/mongo_connector/doc_managers/nodes_and_relationships_builder.py", line 23, in build_nodes_query
    elif self.is_json_array(document[key]):
  File "/Users/lyonwj/neotechnology/devtmp/neo4j_doc_manager/mongo_connector/doc_managers/nodes_and_relationships_builder.py", line 46, in is_json_array
    return ((type(doc_key) is list) and (type(doc_key[0]) is dict))
IndexError: list index out of range
~~~

`mongo-connector.log`:

~~~
2015-09-11 10:07:46,855 [ERROR] mongo_connector.util:87 - Fatal Exception
~~~

**Comment by hannelita** [comment]
@johnymontana Quick question, did you test this before PR 17 was merged? PR#17 implements methods to handle multidimentional arrays.
I have just tested this database and got any import errors.

**Comment by johnymontana** [comment]
@hannelita Tested again with latest version and unable to reproduce.  Closing, fixed in [PR 17](https://github.com/neo4j-contrib/neo4j_doc_manager/pull/17).
